File: backend/models/savings.go
package models

import "time"

type Savings struct {
	ID             string    `gorm:"primaryKey;type:uuid;default:gen_random_uuid()" json:"id"`
	UserID         string    `gorm:"not null;type:uuid;index" json:"user_id"`
	Type           string    `gorm:"default:'general'" json:"type"`
	TargetAmount   float64   `json:"target_amount"`
	CurrentAmount  float64   `gorm:"default:0" json:"current_amount"`
	MonthlyDeposit float64   `json:"monthly_deposit"`
	DueDate        *time.Time `json:"due_date,omitempty"`
	Status         string    `gorm:"default:'active'" json:"status"`
	CreatedAt      time.Time `json:"created_at"`
	UpdatedAt      time.Time `json:"updated_at"`
}

File: server/models/Savings.js
import mongoose from 'mongoose';

const savingsSchema = new mongoose.Schema({
  userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  type: { type: String, enum: ['general', 'dps', 'goal'], default: 'general' },
  targetAmount: { type: Number },
  currentAmount: { type: Number, default: 0 },
  monthlyDeposit: { type: Number },
  dueDate: { type: Date },
  status: { type: String, enum: ['active', 'completed', 'withdrawn'], default: 'active' },
}, { timestamps: true });

export default mongoose.model('Savings', savingsSchema);
